Mathlib/FieldTheory/Perfect.lean

```diff
@@ -218,3 +218,13 @@ instance toPerfectRing (p : ℕ) [hp : Fact p.Prime] [CharP K p] : PerfectRing K
   exact minpoly.degree_pos ha
 
 end PerfectField
+
+/-- If `L / K` is an algebraic extension, `K` is a perfect field, then `L / K` is separable. -/
+theorem Algebra.IsAlgebraic.isSeparable_of_perfectField {K L : Type*} [Field K] [Field L]
+    [Algebra K L] [PerfectField K] (halg : Algebra.IsAlgebraic K L) : IsSeparable K L :=
+  ⟨fun x ↦ PerfectField.separable_of_irreducible <| minpoly.irreducible (halg x).isIntegral⟩
+
+-- TODO:
+/-- If `L / K` is an algebraic extension, `K` is a perfect field, then so is `L`. -/
+proof_wanted Algebra.IsAlgebraic.perfectField {K L : Type*} [Field K] [Field L]
+    [Algebra K L] [PerfectField K] (halg : Algebra.IsAlgebraic K L) : PerfectField L
```

Comment:
I think we have the same result with PerfectField replaced by IsSepClosed ...

Reply:
Not yet, at least loogle can't find it. We only have the result that separably closed field doesn't have non-trivial separable extension.

Reply:
Yeah, sorry for bad wording, I meant it should be true, not that we have it in mathlib. I think the following proof won't be too difficult: assume `IsSepClosed K`, we have `IsPurelyInseparable K (AlgebraicClosure L)` by `isPurelyInseparable_of_isSepClosed` in this PR. By `IsPurelyInseparable.tower_top` we deduce `IsPurelyInseparable L (AlgebraicClosure L)`, which should be equivalent to `separableClosure L (AlgebraicClosure L) = ⊥`, but this is missing from this PR. Finally we use [IsSepClosed.separableClosure_eq_bot_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/SeparableClosure.html#IsSepClosed.separableClosure_eq_bot_iff) to conclude that `IsSepClosed L`.
(We could combine the last steps to a lemma that says `IsSepClosed K` iff `IsPurelyInseparable K (AlgebraicClosure K)`, and dually, `PerfectField K` iff `IsSeparable K (AlgebraicClosure K)`.

Reply:
I have another proof in mind, which uses multiplicativity of (finite) separable degree.

Let me look at your suggestions later.

Reply:
Actually, I think your definition of `IsPurelyInseparable L E` is defeq to `separableClosure L E ≤ ⊥` ...

Reply:
> ... `IsPurelyInseparable L (AlgebraicClosure L)`, which should be equivalent to `separableClosure L (AlgebraicClosure L) = ⊥`, but this is missing from this PR.

It is `separableClosure.eq_bot_iff` at line 182.

Reply:
> We could combine the last steps to a lemma that says `IsSepClosed K` iff `IsPurelyInseparable K (AlgebraicClosure K)`,

I'll add `isSepClosed_iff_isPurelyInseparable_algebraicClosure` later.

> and dually, `PerfectField K` iff `IsSeparable K (AlgebraicClosure K)`.)

For this it seems that I need to use the fact that if `E / F` is separable, `E` is perfect, then `F` is also perfect, which is not in mathlib yet.